src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment.component.ts

```diff
@@ -495,6 +519,10 @@ export class FileUploadAssessmentComponent implements OnInit, AfterViewInit, OnD
         this.referencedFeedback = feedbacks;
     }
 
+    get readOnly(): boolean {
+        return !this.isAtLeastInstructor && !!this.complaint && this.isAssessor;
```

Comment:
This appears multiple times. Would be better if this is defined in one place.

Reply:
There are three assessment components which have very similar code. However, refactoring them to create an assessment superclass would require a deep intrusion into the code which I feel should be done in a separate PR. This change should also include adapting the manual assessment review of programming exercises to follow the same procedure as the other exercises types. However, this is all out of scope of this PR, and we should probably do this after the exams are over.